solutions/CSES Sliding Cost.mdx

```diff
@@ -0,0 +1,100 @@
+---
+id: cses-1077
+source: CSES
+title: Sliding Cost
+author: Isaac Noel
+---
+
+**Time Complexity:** $\mathcal{O}(N\log K)$
+
+## Abstract
+
+Finding the difference between sums of the upper $K/2$ elements and lower $K/2$ with two multisets.
+
+<Warning>
+
+The implementation of this solution assumes familarity with [Sliding Median](/gold/sliding#problem-https://cses.fi/problemset/task/1076) as their implementations are near identical.
+
+</Warning>
+
+## Solution
+
+### The cost of a window
+
+It can be shown that it is optimal to change the values of the window to the median (left as an exercise :D). Once we find the median, we must find the sum of all elements' distance from the median. Rather than finding the absolute value of every individual element's distance, which is far too slow, we can split the window into two groups and use $\mathcal{O}(\log K)$ insertions/deletions to find the cost in constant time. The smallest $K/2$ elements in the window will be in the lower group while the largest $K/2$ elements in the window will be in the upper group.
+
+The cost of the window can be expressed as a function of ${K,S_1,S_2}$, and $M$, where $S_1$ and $S_2$ denote the sum of elements in the lower and upper group respectively, and $M$ denotes the median of the window. The cost of the lower group will be $\sum_{i=1}^{K/2} M-e_i$, and the cost of the upper group will be $\sum_{i=1}^{K/2} e_i-M$, where $e$ represents an element in the group. These expressions can be simplified to $M\times K/2 - S_1$ and $S_2 - M\times K/2$.  After summing cost of the two groups, we have the total cost of the window which is $S_2 - S_1$.
```

Comment:
> ${K,S_1,S_2}$

Why are there curly braces here?

Reply:
> After summing cost of the two groups, we have the total cost of the window which is $S_2 - S_1$.

I think the following is better - The total cost of the window is the sum of the costs contributed by both groups, or $S_2-S_1$.

Reply:
> > ${K,S_1,S_2}$
> 
> Why are there curly braces here?

Didn't realize they were unnecessary, I'll just remove them